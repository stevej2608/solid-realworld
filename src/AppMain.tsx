import { render } from 'solid-js/web'
import { App } from './App'
import { RouterContext, createRouteHandler } from './routeContext'
import { StoreContext, createApplicationStore } from './store/storeContext'

interface IProps {
  children?: Element
}

function ContextProvider(props: IProps) {
  const router = createRouteHandler('')
  const store = createApplicationStore()

  return (
    <RouterContext.Provider value={router}>
      <StoreContext.Provider value={store}>{props.children}</StoreContext.Provider>
    </RouterContext.Provider>
  )
}

export const AppMain = () => (
  <ContextProvider>
    <App />
  </ContextProvider>
)

