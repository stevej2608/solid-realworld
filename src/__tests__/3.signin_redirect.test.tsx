import { HTMLElement } from 'solid-js';
import { render, waitFor, fireEvent } from '@solidjs/testing-library';
import { describe, expect, it } from 'vitest';
import '@testing-library/jest-dom';

import { AppMain } from '../AppMain'

// https://testing-library.com/docs/react-testing-library/cheatsheet/
// https://testing-library.com/docs/guide-disappearance/

describe('Signin redirect', () => {

  // Confirm tag selection opens new tag feed

  it('1. Should see clicking on tag selects a new feed', async () => {
    let result = undefined

    const screen = render(() => <AppMain />)

    expect(screen).toBeTruthy()

    // Confirm we have ten global tags

    result = await screen.findAllByTestId ('tag-link')
    expect((result as HTMLElement[]).length).toBe(10)

    result = await screen.findByTestId ('container')
    expect(result).toMatchSnapshot()

    fireEvent.click(screen.getByText('welcome'))

    await waitFor(() =>
      expect(
        screen.queryByText('Welcome to RealWorld project')
      ).toBeTruthy(), {timeout: 2000} )

    // Final snapshot

    result = await screen.findByTestId ('container')
    expect(result).toMatchSnapshot()

  })

})
