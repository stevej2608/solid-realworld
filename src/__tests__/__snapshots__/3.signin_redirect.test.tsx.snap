// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Signin redirect > 1. Should see clicking on tag selects a new feed 1`] = `
<div
  data-testid="container"
>
  <nav
    class="navbar navbar-light"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="#/"
      >
        conduit
      </a>
      <ul
        class="nav navbar-nav pull-xs-right"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="#/"
          >
            Home
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="#/login"
          >
            Sign in
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="#/register"
          >
            Sign up
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    class="home-page"
  >
    <div
      class="banner"
    >
      <div
        class="container"
      >
        <h1
          class="logo-font"
        >
          conduit
        </h1>
        <p>
          A place to share your knowledge.
        </p>
      </div>
    </div>
    <div
      class="container page"
    >
      <div
        class="row"
      >
        <div
          class="col-md-9"
        >
          <div
            class="feed-toggle"
          >
            <ul
              class="nav nav-pills outline-active"
            >
              
              <li
                class="nav-item"
              >
                <a
                  class="nav-link active"
                  href="#/?tab=all"
                >
                  Global Feed
                </a>
              </li>
              
            </ul>
          </div>
          <div
            class="article-preview"
          >
            Loading articles...
          </div>
        </div>
        <div
          class="col-md-3"
        >
          <div
            class="sidebar"
          >
            <p>
              Popular Tags
            </p>
            <div
              class="tag-list"
            >
              <a
                class="tag-pill tag-default"
                data-testid="tag-link"
                href="#/?tab=implementations"
              >
                implementations
              </a>
              <a
                class="tag-pill tag-default"
                data-testid="tag-link"
                href="#/?tab=welcome"
              >
                welcome
              </a>
              <a
                class="tag-pill tag-default"
                data-testid="tag-link"
                href="#/?tab=introduction"
              >
                introduction
              </a>
              <a
                class="tag-pill tag-default"
                data-testid="tag-link"
                href="#/?tab=codebaseshow"
              >
                codebaseshow
              </a>
              <a
                class="tag-pill tag-default"
                data-testid="tag-link"
                href="#/?tab=ipsum"
              >
                ipsum
              </a>
              <a
                class="tag-pill tag-default"
                data-testid="tag-link"
                href="#/?tab=qui"
              >
                qui
              </a>
              <a
                class="tag-pill tag-default"
                data-testid="tag-link"
                href="#/?tab=cupiditate"
              >
                cupiditate
              </a>
              <a
                class="tag-pill tag-default"
                data-testid="tag-link"
                href="#/?tab=et"
              >
                et
              </a>
              <a
                class="tag-pill tag-default"
                data-testid="tag-link"
                href="#/?tab=quia"
              >
                quia
              </a>
              <a
                class="tag-pill tag-default"
                data-testid="tag-link"
                href="#/?tab=deserunt"
              >
                deserunt
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Signin redirect > 1. Should see clicking on tag selects a new feed 2`] = `
<div
  data-testid="container"
>
  <nav
    class="navbar navbar-light"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href="#/"
      >
        conduit
      </a>
      <ul
        class="nav navbar-nav pull-xs-right"
      >
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="#/"
          >
            Home
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="#/login"
          >
            Sign in
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-link"
            href="#/register"
          >
            Sign up
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    class="home-page"
  >
    <div
      class="banner"
    >
      <div
        class="container"
      >
        <h1
          class="logo-font"
        >
          conduit
        </h1>
        <p>
          A place to share your knowledge.
        </p>
      </div>
    </div>
    <div
      class="container page"
    >
      <div
        class="row"
      >
        <div
          class="col-md-9"
        >
          <div
            class="feed-toggle"
          >
            <ul
              class="nav nav-pills outline-active"
            >
              
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="#/?tab=all"
                >
                  Global Feed
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link active"
                  href=""
                >
                  <i
                    class="ion-pound"
                  />
                   
                  welcome
                </a>
              </li>
            </ul>
          </div>
          <div
            class="article-preview"
          >
            <div
              class="article-meta"
            >
              <a
                href="#/@Gerome"
              >
                <img
                  alt=""
                  src="https://api.realworld.io/images/demo-avatar.png"
                />
              </a>
              <div
                class="info"
              >
                <a
                  class="author"
                  href="#/@Gerome"
                >
                  Gerome
                </a>
                <span
                  class="date"
                >
                  Wed Nov 24 2021
                </span>
              </div>
              <div
                class="pull-xs-right"
              >
                <button
                  class="btn btn-sm btn-outline-primary"
                >
                  <i
                    class="ion-heart"
                  />
                  Â  
                  
                  <!---->
                   
                  <span
                    class="counter"
                  >
                    (
                    3194
                    <!---->
                    )
                  </span>
                </button>
              </div>
            </div>
            <a
              class="preview-link"
              href="#/article/Welcome-to-RealWorld-project-1"
            >
              <h1>
                Welcome to RealWorld project
              </h1>
              <p>
                Exemplary fullstack Medium.com clone powered by React, Angular, Node, Django, and many more
              </p>
              <span>
                Read more...
              </span>
              <ul
                class="tag-list"
              >
                <li
                  class="tag-default tag-pill tag-outline"
                >
                  welcome
                </li>
                <li
                  class="tag-default tag-pill tag-outline"
                >
                  introduction
                </li>
              </ul>
            </a>
          </div>
          <nav>
            <ul
              class="pagination"
            >
              <li
                class="page-item active"
              >
                <a
                  class="page-link"
                  href=""
                >
                  1
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href=""
                >
                  2
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href=""
                >
                  3
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href=""
                >
                  4
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href=""
                >
                  5
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href=""
                >
                  6
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href=""
                >
                  7
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href=""
                >
                  8
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href=""
                >
                  9
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href=""
                >
                  10
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href=""
                >
                  11
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href=""
                >
                  12
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href=""
                >
                  13
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href=""
                >
                  14
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href=""
                >
                  15
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href=""
                >
                  16
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href=""
                >
                  17
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href=""
                >
                  18
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href=""
                >
                  19
                </a>
              </li>
              <li
                class="page-item"
              >
                <a
                  class="page-link"
                  href=""
                >
                  20
                </a>
              </li>
            </ul>
          </nav>
        </div>
        <div
          class="col-md-3"
        >
          <div
            class="sidebar"
          >
            <p>
              Popular Tags
            </p>
            <div
              class="tag-list"
            >
              <a
                class="tag-pill tag-default"
                data-testid="tag-link"
                href="#/?tab=implementations"
              >
                implementations
              </a>
              <a
                class="tag-pill tag-default"
                data-testid="tag-link"
                href="#/?tab=welcome"
              >
                welcome
              </a>
              <a
                class="tag-pill tag-default"
                data-testid="tag-link"
                href="#/?tab=introduction"
              >
                introduction
              </a>
              <a
                class="tag-pill tag-default"
                data-testid="tag-link"
                href="#/?tab=codebaseshow"
              >
                codebaseshow
              </a>
              <a
                class="tag-pill tag-default"
                data-testid="tag-link"
                href="#/?tab=ipsum"
              >
                ipsum
              </a>
              <a
                class="tag-pill tag-default"
                data-testid="tag-link"
                href="#/?tab=qui"
              >
                qui
              </a>
              <a
                class="tag-pill tag-default"
                data-testid="tag-link"
                href="#/?tab=cupiditate"
              >
                cupiditate
              </a>
              <a
                class="tag-pill tag-default"
                data-testid="tag-link"
                href="#/?tab=et"
              >
                et
              </a>
              <a
                class="tag-pill tag-default"
                data-testid="tag-link"
                href="#/?tab=quia"
              >
                quia
              </a>
              <a
                class="tag-pill tag-default"
                data-testid="tag-link"
                href="#/?tab=deserunt"
              >
                deserunt
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
